coalib/settings/Setting.py

```diff
@@ -296,3 +297,26 @@ def value(self):
                              'incomplete. Please access the value of the '
                              'setting in a section to get the complete value.')
         return self._value
+
+    @property
+    def origin(self):
+        """
+        Returns the filename.
+        """
+        if isinstance(self._origin, SourcePosition):
+            return self._origin.filename
+        else:
+            return self._origin
+
+    @origin.setter
```

Comment:
no, we need to remove this.

it is extremely bad practise for `a.value = a.value` to have any effect.

think about it.

if the property getter always returns a string filename, then if the caller uses the setter with a `SourcePosition`, the object will be lost.

```py
sp = SourcePosition('foo.txt',...)
s = Setting(origin=sp,...)
print(type(s.origin))
filename = s.origin  # a string for backwards compatibility
s.origin = filename
print(type(s.origin))
```

Reply:
yup. unfortunately , we didnt think about this when we suggested to add a setter.

Reply:
this hasnt been done.

Reply:
If setter is not there then on running pytest, a test is failing.
In docstring of `get_config_directory` in `ConfigurationGathering.py` there is a line (line number 371) which sets the `origin` and if setter is not there then error is shown `AttributeError: can't set attribute`.

Reply:
That doctest needs to be fixed.  It was showing what was possible, by creating a ficticious scenario.
It was cheating.  It needs to be fixed.